controllers/cluster_controller_test.go

```diff
@@ -115,6 +118,40 @@ func TestClusterReconcilerReconcileSelfManagedCluster(t *testing.T) {
 	g.Expect(result).To(Equal(ctrl.Result{}))
 }
 
+func TestClusterReconcilerSelfManagedClusterWithMatchingGenerations(t *testing.T) {
```

Comment:
could we add child objs to this test? That makes it more realistic. IRL we will never reconcile a cluster without linked objects

Also, I would add tests for the different no matching generations:
- same generation, diff aggregated generation
- diff aggregation, same aggregated generation
- diff aggregation, diff aggregated generation

this can easily be a table test

Reply:
Will update!